### PR TITLE
dataset should not accept upper case letter

### PR DESCRIPTION
**Pull request by saschanaz**
http://www.w3.org/TR/2012/CR-html5-20121217/dom.html#embedding-custom-non-visible-data-with-the-data-*-attributes
'2.For each content attribute on the element whose first five characters are the string "data-" and whose remaining characters (if any) do not include any characters in the range U+0041 to U+005A (LATIN CAPITAL LETTER A to LATIN CAPITAL LETTER Z)'

### CODE REVIEW

File: tests/dataset/test.js [hunk 1]
```diff
@@ -68,10 +68,24 @@ test("Dataset Get*", function() {
       var fixture = document.createElement("div"),
           value = index + "";
 
-      fixture.setAttribute( array[0], index + "" );
+      fixture.setAttribute( array[0], value );
```
Comment (rwaldron) [inline tests/dataset/test.js:71]: This fix is good.

File: tests/dataset/test.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ test("Dataset Get*", function() {
       [ "data--", "-" ],
       [ "data--foo", "Foo" ],
       [ "data---foo", "-Foo" ],
-      [ "data-Foo", "foo" ],
+      //[ "data-Foo", "foo" ],
```
Comment (rwaldron) [inline tests/dataset/test.js]: All of the cases here are valid.

File: tests/dataset/test.js [hunk 3]
```diff
@@ -68,10 +68,24 @@ test("Dataset Get*", function() {
...
+
+      var fixture = document.createElement("div"),
+          value = index + "",
+          result;
+
+      fixture.setAttribute(str, value);
+      for (var s in fixture.dataset)
+        result = s;
+      
+      assert( result === undefined, "Expected: undefined" );
```
Comment (rwaldron) [inline tests/dataset/test.js]: This is wrong. 

``` js
var div = document.createElement("div"); 
div.setAttribute("data-Foo", 1); 
(div.dataset.foo === "1") === true;
```

data-\* properties are always lower cased and all values in dataset become strings (conversion per [DOMString](http://www.w3.org/TR/2012/CR-html5-20121217/infrastructure.html#domstringmap-0))

Comment (saschanaz) [inline tests/dataset/test.js]: @rwaldron Yes, data-\* properties should be always lower cased, but in this case, data-Foo should not be recognized as data-\* property, as the algorithm documented.

Comment (rwaldron) [inline tests/dataset/test.js]: > All attribute names on HTML elements in HTML documents get ASCII-lowercased automatically, so the restriction on ASCII uppercase letters doesn't affect such documents.

Comment (saschanaz) [inline tests/dataset/test.js]: Oh, I got the point, thank you. I'll remove those codes, except one :)

**Comment by saschanaz** [comment]
Roll backed!

**Pull request closed without merging**